### PR TITLE
make some private rsync, and exec_cluster arguments public

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?

These arguments unlock customizations for rsync and run_on_cluster for anyone using the autoscaler public api.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]

File: python/ray/autoscaler/sdk.py [hunk 1]
```diff
@@ -125,6 +130,8 @@ def rsync(cluster_config: Union[dict, str],
             public or private.
         no_config_cache (bool): Whether to disable the config cache and fully
             resolve all environment settings from the Cloud provider again.
+        all_nodes (bool): whether to sync worker nodes in addition to the head
```
Comment (pcmoritz) [inline python/ray/autoscaler/sdk.py:133]: Didn't we want to deprecate this because of the race condition between rsync and starting up more workers?

Comment (alanwguo) [reply python/ray/autoscaler/sdk.py:133]: Can you explain more? I see we use it in the product and it seems like we do want to sync files to all the workers, what's the alternative to this?

Comment (ijrsvt) [reply python/ray/autoscaler/sdk.py:133]: @ericl has previously expressed interest in keeping this private:
https://github.com/ray-project/ray/pull/11517#discussion_r509825458

Comment (alanwguo) [reply python/ray/autoscaler/sdk.py:133]: @ericl , is the goal to deprecate all_nodes in the `ray CLI`? If so, I can remove it as an option in the anyscale CLI

Comment (ericl) [reply python/ray/autoscaler/sdk.py:133]: Yes, we will remove this option.

Comment (ericl) [reply python/ray/autoscaler/sdk.py:133]: Thanks for catching this @pcmoritz ... somehow I missed it in first review.

**Review by ericl — CHANGES_REQUESTED** [review]

**Review by ijrsvt — APPROVED** [review]
Looks good now =)

**Review by ericl — APPROVED** [review]

**Pull request merged**